### DISCUSSION TITLE
Cannot bind argument to parameter 'AzureCloud' because it is an empty string.

### DISCUSSION THREAD

**Discussion by jonprattC78** [Azure/ipam#46] [Q&A] [answered]
I have a issue when I try to deploy the IPAM "Apps Only" option (./deploy.ps1 - AppsOnly)

The logged error is:

Deploy-IPAMApplications: /Users/jon.pratt@domain.io/IaC/az-ipam/deploy/deploy.ps1:757:19
Line |
 757 |        -AzureCloud $azureCloud `
     |                    ~~~~~~~~~~~
     | Cannot bind argument to parameter 'AzureCloud' because it is an empty string.

I am following these steps:
Connect-AzAccount
Get-AzSubscription
Set-AzContext -subscription "xxxx-xxxx-xxxx-xxxx"
then run ./deploy.ps1 - AppsAonly

then it bombs out with the above error.
I was thinking about hard coding the $azureCloud parameter in the script - but I dont understand why the FULL application deployment does not have this same issue?

**Comment by DCMattyG** [comment]
Hi @jonprattC78, thank you so much for bringing this to my attention. I see that I have an issue in the Deploy script that is causing this. I'll push a PR in the morning to address this issue.

**Comment by jonprattC78** [comment]
Ok – thanks Matty for this …


From: DCMattyG ***@***.***>
Date: Monday, 7 November 2022 at 09:47
To: Azure/ipam ***@***.***>
Cc: jonprattC78 ***@***.***>, Mention ***@***.***>
Subject: Re: [Azure/ipam] Cannot bind argument to parameter 'AzureCloud' because it is an empty string. (Discussion #46)

Hi @jonprattC78<https://www.google.com/url?q=https://github.com/jonprattC78&source=gmail-imap&ust=1668419224000000&usg=AOvVaw34wVijsognbpWAYL3cyrOC>, thank you so much for bringing this to my attention. I see that I have an issue in the Deploy script that is causing this. I'll push a PR in the morning to address this issue.

—
Reply to this email directly, view it on GitHub<https://www.google.com/url?q=https://github.com/Azure/ipam/discussions/46%23discussioncomment-4074785&source=gmail-imap&ust=1668419224000000&usg=AOvVaw20UIf6AyG8qVFJl-Ri-tNf>, or unsubscribe<https://www.google.com/url?q=https://github.com/notifications/unsubscribe-auth/A2Q6OH4445QL7ZI7KHND523WHC6YPANCNFSM6AAAAAARY7TVUE&source=gmail-imap&ust=1668419224000000&usg=AOvVaw2iAwBO8LIGKpi1NZzMsEOj>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

-- 
CONTINO


This e-mail, and any attachment, is confidential. If you have 
received it in error, please delete it from your system, do not use or 
disclose the information in any way, and notify me immediately. The 
contents of this message may contain personal views, which are not the 
views of industry, unless specifically stated.


https://www.contino.io 
<http://www.contino.io>

**Comment by DCMattyG** [comment, marked as answer]
Hey @jonprattC78, I just merged the fixes so please pull the latest code from GitHub, give it a try, and let me know if you have any further issues. Happy to address any other concerns you may have!

**Comment by jonprattC78** [comment]
Hey!

Just to let you know, works as expected in -AppsOnly mode!

Thanks for your support as always!

Jon


From: DCMattyG ***@***.***>
Date: Tuesday, 8 November 2022 at 05:16
To: Azure/ipam ***@***.***>
Cc: jonprattC78 ***@***.***>, Mention ***@***.***>
Subject: Re: [Azure/ipam] Cannot bind argument to parameter 'AzureCloud' because it is an empty string. (Discussion #46)

Hey @jonprattC78<https://www.google.com/url?q=https://github.com/jonprattC78&source=gmail-imap&ust=1668489407000000&usg=AOvVaw3wDn6SNsRmP0PVtNA44d5c>, I just merged the fixes so please pull the latest code from GitHub, give it a try, and let me know if you have any further issues. Happy to address any other concerns you may have!

—
Reply to this email directly, view it on GitHub<https://www.google.com/url?q=https://github.com/Azure/ipam/discussions/46%23discussioncomment-4082951&source=gmail-imap&ust=1668489407000000&usg=AOvVaw3rOxOkEHtke4_wWDQ3KJQn>, or unsubscribe<https://www.google.com/url?q=https://github.com/notifications/unsubscribe-auth/A2Q6OH2RQOKVL3ORPO63WSDWHHO3ZANCNFSM6AAAAAARY7TVUE&source=gmail-imap&ust=1668489407000000&usg=AOvVaw3vnTJ-VzgIfM9boW2u4gI4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

-- 
CONTINO


This e-mail, and any attachment, is confidential. If you have 
received it in error, please delete it from your system, do not use or 
disclose the information in any way, and notify me immediately. The 
contents of this message may contain personal views, which are not the 
views of industry, unless specifically stated.


https://www.contino.io 
<http://www.contino.io>
